ecl/eclcmd/eclcmd_core.cpp

```diff
@@ -1959,15 +1964,38 @@ class EclCmdZapGen : public EclCmdCommon
             if ((strstr(errMsg.str(), "Warning") != nullptr) || (strstr(errMsg.str(), "Failed") != nullptr))
             {
                 fprintf(stderr, "Zapgen command returns with error:\n%s\n", errMsg.str());
-                return false;
+                isOk = false;
+            }
+            else if ((strstr(errMsg.str(), "< Location") != nullptr) && (strstr(errMsg.str(), "Login.html") != nullptr))
+            {
+                fprintf(stderr, "Zapgen command needs to authenticate, please provide credentials.\n");
+                isOk = false;
+            }
+            else if (strstr(errMsg.str(), "Authentication problem") != nullptr)
+            {
+                fprintf(stderr, "\nZapgen command returns with authentication problem. Please check your credentials.\n");
+                isOk = false;
+                removeOutFile = true; // It contains the authentication error message only
             }
-            else if (optVerbose)
+
+            if (optVerbose)
                 fprintf(stderr, "Zapgen command returns with:\n%s\n", errMsg.str());
 
         }
 
-        fprintf(stdout, "ZAP file written into %s.\n", outputFile.str());
-        return 0;
+        if (isOk)
+        {
+            fprintf(stdout, "ZAP file written into %s.\n", outputFile.str());
+            return 0;
+        }
+        else
+        {
+            Owned<IFile> file = createIFile(outputFile.str());
+            if (file->exists() && (removeOutFile || (0 == file->size())))
+                file->remove();
+            return false;
```

Comment:
Return 0 above, false here... return type is int, should this be a number?

Reply:
Changed